### ISSUE TITLE
BufferUnderflowException when using SaveReader.read(file)

### ISSUE THREAD

**Issue by TJPlaysNow** [Memzer/RailroadsOnlineSaveEditor#1]
Getting this error in my code.

```
java.nio.BufferUnderflowException
	at java.base/java.nio.HeapByteBuffer.get(HeapByteBuffer.java:183)
	at java.base/java.nio.ByteBuffer.get(ByteBuffer.java:822)
	at com.rr.rrol.se.io.BinaryReader.readString(BinaryReader.java:105)
	at com.rr.rrol.se.model.property.item.TextItem.<init>(TextItem.java:43)
	at com.rr.rrol.se.model.property.ArrayProperty.<init>(ArrayProperty.java:80)
	at com.rr.rrol.se.model.property.PropertyFactory.read(PropertyFactory.java:46)
	at com.rr.rrol.se.io.SaveReader.read(SaveReader.java:69)
	at dev.tjplaysnow.choochoo.main.GameSave.generate(GameSave.java:22)
	at dev.tjplaysnow.choochoo.main.GameSave.<init>(GameSave.java:16)
	at dev.tjplaysnow.choochoo.main.Main.main(Main.java:31)
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "dev.tjplaysnow.choochoo.main.GameSave.getSave()" because "dev.tjplaysnow.choochoo.main.Main.gameSave" is null
	at dev.tjplaysnow.choochoo.main.Main.main(Main.java:36)
```

```
System.out.println(System.getenv("LOCALAPPDATA") + "\\arr\\Saved\\SaveGames\\slot" + saveNumber + ".sav");
InputStream fis = new FileInputStream(System.getenv("LOCALAPPDATA") + "\\arr\\Saved\\SaveGames\\slot" + saveNumber + ".sav");
save = SaveReader.read(fis);
```

This is the code I am running which gives me this error on the bottom line.
The error internal points to a weird line, in the built api it looks like this and is the middle line in the for loop

```
for(Property property = PropertyFactory.read(reader); property != null; property = PropertyFactory.read(reader)) {
    properties.add(property);
}
```

Although the line in the actual code looks like this:

```
List<Property> properties = new ArrayList<>();
@SuppressWarnings("rawtypes")
Property property = PropertyFactory.read(reader);
while(property != null) {
	properties.add(property);
	property = PropertyFactory.read(reader);
}
```

This being said it seems the error falls all the way to:

```
if(length > 0) {
	bb.get(valueBytes);
	String s = new String(valueBytes);
	return s.substring(0, s.length()-1);
} else {
```

with the line `bb.get(valueBytes);` being the problem.

**Comment by Memzer** [comment]
Thank you for the report. This appears to be an issue reading a string value from the save file. Perhaps some strange unicode characters in a player name?

If you can attach your save file here, I'll try to reproduce and debug the issue. Thanks!

**Comment by TJPlaysNow** [comment]
[slot1.zip](https://github.com/Memzer/RailroadsOnlineSaveEditor/files/7523597/slot1.zip)
It is most likely to do with a player name, have had quite a few different players on the game so far.

**Comment by Memzer** [comment]
Actually this was not a player name issue. The cause was from a name given to a vehicle, where the first part of the in-game formatter was null. I'm surprised the game allows this!

I've pushed the fix, and incremented the version to 0.0.2-SNAPSHOT
